src/main/java/pl/allegro/tech/build/axion/release/infrastructure/git/GitRepository.java

```diff
@@ -333,37 +304,54 @@ private void assertPathExists(String path) {
     }
 
     public ScmPosition currentPosition() {
+        String revision = getRevision();
+        String branchName = branchName();
+        boolean isClean = !checkUncommittedChanges();
+        return new ScmPosition(revision, branchName, isClean);
+    }
+
+    private String getRevision() {
         try {
             String revision = "";
             if (hasCommits()) {
                 ObjectId head = head();
                 revision = head.name();
             }
-
-            boolean isClean = !checkUncommittedChanges();
-
-            String branchName = branchName();
-            return new ScmPosition(revision, branchName, isClean);
+            return revision;
         } catch (IOException e) {
             throw new ScmException(e);
         }
     }
 
+    private String branchName() {
+        return branchNameFromEnvVariable().orElseGet(this::branchNameFromGit);
```

Comment:
@radoslaw-panuszewski could you add docs entry about this happening?

Reply:
Done